gapis panic results in UI exception.

The gapis panic is never shown. :(

```
org.eclipse.swt.SWTException: Failed to execute runnable (java.lang.reflect.UndeclaredThrowableException)
	at org.eclipse.swt.SWT.error(Unknown Source)
	at org.eclipse.swt.SWT.error(Unknown Source)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Unknown Source)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Unknown Source)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Unknown Source)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at com.google.gapid.Main$UI.show(Main.java:109)
	at com.google.gapid.Main.main(Main.java:67)
Caused by: java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy7.onStateLoaded(Unknown Source)
	at com.google.gapid.models.ApiState.fireLoadedEvent(ApiState.java:113)
	at com.google.gapid.models.ModelBase.updateError(ModelBase.java:106)
	at com.google.gapid.models.ApiState.updateError(ApiState.java:102)
	at com.google.gapid.models.ApiState.updateError(ApiState.java:49)
	at com.google.gapid.models.ModelBase$1.onUiThreadError(ModelBase.java:79)
	at com.google.gapid.rpc.UiErrorCallback.onUiThread(UiErrorCallback.java:36)
	at com.google.gapid.rpc.UiErrorCallback.onUiThread(UiErrorCallback.java:25)
	at com.google.gapid.rpc.UiCallback.lambda$onFinish$0(UiCallback.java:45)
	at com.google.gapid.widgets.Widgets.ifNotDisposed(Widgets.java:118)
	at com.google.gapid.widgets.Widgets.lambda$scheduleIfNotDisposed$1(Widgets.java:146)
	at com.google.gapid.widgets.Widgets.lambda$schedule$0(Widgets.java:138)
	at org.eclipse.swt.widgets.RunnableLock.run(Unknown Source)
	... 7 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gapid.util.Events$ListenerCollectionImpl.invoke(Events.java:157)
	... 20 more
Caused by: org.eclipse.core.runtime.AssertionFailedException: null argument:
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:85)
	at org.eclipse.core.runtime.Assert.isNotNull(Assert.java:73)
	at org.eclipse.jface.viewers.TreePath.<init>(TreePath.java:47)
	at com.google.gapid.views.StateView.onStateLoaded(StateView.java:335)
	... 25 more
```

--- a/gapic/src/main/com/google/gapid/models/ApiState.java
+++ b/gapic/src/main/com/google/gapid/models/ApiState.java
@@ -19,6 +19,7 @@ import static com.google.gapid.rpc.UiErrorCallback.error;
 import static com.google.gapid.rpc.UiErrorCallback.success;
 import static com.google.gapid.util.Paths.stateTree;
 import static com.google.gapid.widgets.Widgets.submitIfNotDisposed;
+import static java.util.logging.Level.SEVERE;
 import static java.util.logging.Level.WARNING;
 
 import com.google.common.util.concurrent.Futures;
@@ -90,17 +91,16 @@ public class ApiState
       LOG.log(WARNING, "Failed to load the API state", e);
       return error(Loadable.Message.error(e));
     } catch (ExecutionException e) {
-      return super.processResult(result);
+      if (!shell.isDisposed()) {
+        LOG.log(SEVERE, "Failed to load the API state", e);
+      }
+      return error(Loadable.Message.error("Failed to load the state"));
     }
   }
 
   @Override
   protected void updateError(Loadable.Message error) {
-    if (error != null) {
-      listeners.fire().onStateLoaded(error);
-    } else {
-      super.updateError(error);
-    }
+    listeners.fire().onStateLoaded(error);
   }
 
   @Override

